editor/scene_tree_dock.cpp

```diff
@@ -2059,6 +2059,8 @@ void SceneTreeDock::_selection_changed() {
 		_tool_selected(TOOL_MULTI_EDIT);
 	} else if (selection_size == 0) {
 		editor->push_item(nullptr);
+	} else {
+		editor->push_item(EditorNode::get_singleton()->get_editor_selection()->get_selected_node_list()[0]);
```

Comment:
I wonder if that's the proper fix. Currently changing selection from one node to another works fine, so it means this new `push_item` would be redundant when going from single selection to single selection.

Reply:
Ah I see. But how we can tell to the editor we are swapping? I think this need more rework than a simple couple of lines. Like... changing how editor for MultiNodeEdit works, no?

Reply:
Seems fine to me, it is also used in CanvasItemEditor:

https://github.com/godotengine/godot/blob/a9c53fa5994db8badb4ec14581bd8422fb340f41/editor/plugins/canvas_item_editor_plugin.cpp#L726-L728

Reply:
I think I found the fix, and submitted PR #49725. Don't mean to step on any toes - I just found some other bugs at the same time so tried to fix those as well.

See here for my solution to this issue:
https://github.com/godotengine/godot/pull/49725/files#diff-80799453dbf9f78a65dda9407ff4c117de273e621d485eca54703dffbcff2de1R662-R669

Reply:
I think then we can close mine. I like your approach, let's see what Akien says :)